src/mgr/PyModuleRegistry.cc

```diff
@@ -67,6 +67,10 @@ void PyModuleRegistry::init()
   std::list<std::string> failed_modules;
 
   std::set<std::string> module_names = probe_modules();
+  if (module_names.empty()) {
+    // This should never happen, but will if mgr_module_path is set incorrectly
+    clog->error() << "No ceph-mgr modules found";
+  }
```

Comment:
i'd suggest move this hunk down after we load the python modules. i.e., to put it before the `if (!failed_modules.empty()) {` block. and might want to refactor `probe_modules()` to pass the path to it, and include the path in the log message, like
```c++
const std::string module_path = g_conf().get_val<std::string>("mgr_module_path");
for (const auto& module_name : probe_modules(module_path)) {
 // ...
}
if (modules.empty()) {
  clog->error() << "No ceph-mgr modules loaded in " << module_path;
}
if (!failed_modules.empty()) {
  // ...
}
```

Reply:
Yeah, that's a bit neater, thanks :-)